internal/resources/managementclusterregistration/resource_management_cluster_registration.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package managementclusterregistration
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	clustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster"
+	managementclusterregistrationmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/managementclusterregistration"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func ResourceManagementClusterRegistration() *schema.Resource {
+	return &schema.Resource{
+		Schema: managementClusterRegistrationSchema,
+		//CreateContext: , TODO
+		//ReadContext: , TODO
+		//UpdateContext: , TODO
+		//DeleteContext: , TODO
+		Description: "Tanzu Mission Control Management Cluster Registration Resource",
+	}
+}
+
+var managementClusterRegistrationSchema = map[string]*schema.Schema{
+	NameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of this cluster",
+		Required:    true,
+		ForceNew:    true,
+	},
+	common.MetaKey:   common.Meta,
+	specKey:          managementClusterRegistrationSpecSchema,
+	attachClusterKey: attachCluster,
+	StatusKey: {
+		Type:        schema.TypeMap,
+		Description: "Status of the cluster",
+		Computed:    true,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+	},
+	waitKey: {
+		Type:        schema.TypeString,
+		Description: "Wait timeout duration until cluster resource reaches READY state. Accepted timeout duration values like 5s, 45m, or 3h, higher than zero. Should be set to 0 in case of simple attach cluster where kubeconfig input is not provided.",
+		Default:     "default",
+		Optional:    true,
+	},
+}
+
+var attachCluster = &schema.Schema{
```

Comment:
Are we supporting creation of workload cluster along with management clusters?
We cannot create a workload cluster until the management cluster comes up healthy. Therefore, I think we cannot have both together in the same resource. Instead we can refer cluster resource to the management cluster resource.

cc: @vaamarnath @ramya-bangera

Reply:
I updated the naming, so hopefully there is less confusion now.

Reply:
Thank you @ekoop-terasky, sorry for the confusion due to the naming.